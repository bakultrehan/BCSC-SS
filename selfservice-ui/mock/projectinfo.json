{
  "project-info": [
    {
      "id": 1,
      "organizationName": "organic",
      "projectName": "project",
      "myRole": "1",
      "developerDetails": {
        "email": "test@test.com",
        "phone": "0987654321",
        "firstName": "name",
        "lastName": "l name",
        "id": 1
      },
      "managerDetails": {
        "email": "mail@test.com",
        "phone": "1234567890",
        "firstName": "f manager",
        "lastName": "l manager",
        "id": 2
      },
      "ctoDetails": {
        "email": "cto@email.com",
        "phone": "1234567890",
        "firstName": "f cto",
        "lastName": "l cto",
        "id": 3
      }
    },
    {
      "organizationName": "aa",
      "projectName": "aa",
      "description": "testaa",
      "myRole": "1",
      "developerDetails": {},
      "managerDetails": {
        "firstName": "mf",
        "lastName": "ml",
        "email": "me",
        "phone": "mp"
      },
      "ctoDetails": {
        "firstName": "cf",
        "lastName": "cl",
        "email": "ce",
        "phone": "cp"
      },
      "id": 11
    },
    {
      "organizationName": "ss1",
      "projectName": "dd",
      "description": "test",
      "myRole": "1",
      "developerDetails": {},
      "managerDetails": {
        "firstName": "mf",
        "lastName": "m",
        "email": "m",
        "phone": "m"
      },
      "ctoDetails": {
        "firstName": "c",
        "lastName": "c",
        "email": "c",
        "phone": "c"
      },
      "id": 12
    }
  ]
}