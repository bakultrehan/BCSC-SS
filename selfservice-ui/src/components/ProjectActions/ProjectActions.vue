/** * Dashboard of app */

<template>
  <div>
    <v-col
      cols="12"
      class="d-flex justify-end"
      v-if="projectStatus === projectStatusList.development"
      no-gutters
    >
      <v-card flat>
        <Button
          @click="toggleWarning()"
          :aria-label="$t('projectActions.btnRequestLiveAccess')"
          class="btn-req"
          secondary
        >{{ $t('projectActions.btnRequestLiveAccess') }}</Button>
      </v-card>
    </v-col>

    <v-col
      class="d-flex justify-end btn-delete pad-0"
      @click="toggleDelete()"
      v-if="isAdmin || projectStatus < projectStatusList.developmentComplete"
      no-gutters
    >
      <v-icon class="ml-2 icon-delete" small>mdi-delete</v-icon>
      {{ $t('projectActions.labelDelete') }}
    </v-col>

    <v-col v-if="requestDialog || deleteDialog">
      <v-dialog v-model="requestDialog" persistent max-width="70%">
        <v-card>
          <v-toolbar flat class="bc-subtitle" dark>
            <v-toolbar-title>
              {{
              $t('projectActions.requestLiveAccessDialogTitle')
              }}
            </v-toolbar-title>
          </v-toolbar>
          <v-card-text
            class="text-left ma-8"
            v-html="$t('projectActions.requestLiveAccessDialogInfo')"
          ></v-card-text>
          <v-card-actions class="pb-10">
            <v-spacer></v-spacer>
            <Button secondary text @click="toggleWarning()">
              {{
              $t('projectActions.btnCancel')
              }}
            </Button>
            <Button text class="btn-live" @click="confirmLiveAccess()">
              {{
              $t('projectActions.btnConfirm')
              }}
            </Button>
          </v-card-actions>
        </v-card>
      </v-dialog>

      <v-dialog v-model="deleteDialog" persistent max-width="70%">
        <v-card>
          <v-toolbar flat class="bc-subtitle" dark>
            <v-toolbar-title>
              <v-icon class="ml-2" medium>mdi-delete</v-icon>
              {{ $t('projectActions.deleteDialogTitle') }}
            </v-toolbar-title>
          </v-toolbar>
          <v-card-text class="text-left ma-8" v-html="$t('projectActions.deleteDialogInfo')"></v-card-text>
          <v-card-actions class="pb-10">
            <v-spacer></v-spacer>
            <Button secondary text @click="toggleDelete()">
              {{
              $t('projectActions.btnDeleteCancel')
              }}
            </Button>
            <Button
              text
              class="dialog-delete"
              @click="confirmDeleteProject()"
            >{{ $t('projectActions.btnDeleteConfirm') }}</Button>
          </v-card-actions>
        </v-card>
      </v-dialog>
    </v-col>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop, Watch } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';

import Button from '@/Atomic/Button/Button.vue';
import { projectStatus } from '@/constants/enums';

const ProjectInfoModule = namespace('ProjectInfoModule');
const KeyCloakModule = namespace('KeyCloakModule');

@Component({
  components: {
    Button
  }
})
export default class ProjectActions extends Vue {
  @Prop({ default: 0 })
  public id!: number;
  @ProjectInfoModule.Getter('getChangeStatus')
  public getChangeStatus!: any;
  @ProjectInfoModule.Action('updateProjectStatus')
  public updateProjectStatus!: any;

  @ProjectInfoModule.Getter('getDeleteProjectReturn')
  public getDeleteProjectReturn!: any;
  @ProjectInfoModule.Action('deleteProject')
  public deleteProject!: any;

  @ProjectInfoModule.Getter('getSingleProjectInfo')
  public getSingleProjectInfo!: any;

  @KeyCloakModule.Getter('isAdmin')
  private isAdmin!: any;

  private requestDialog: boolean = false;
  private deleteDialog: boolean = false;

  private showProjectActions: boolean = false;
  private projectStatus: number = 1;
  private projectStatusList: any = projectStatus;

  @Watch('getChangeStatus')
  private ongetChangeStatusChanged(val: any) {
    const { statusChangeError, statusChangeSuccess } = val;
    this.toggleWarning();
    if (statusChangeSuccess === true) {
      this.$router.push(`/project-container/${this.id}`);
    }
  }
  @Watch('getDeleteProjectReturn')
  private ongetgetDeleteProjectReturnChanged(val: any) {
    const { deleteProjectError, deleteProjectSuccess } = val;

    if (deleteProjectSuccess === true) {
      this.$router.push('/dashboard');
    }
  }

  /** this watch will get called on load of projectinfo summary page which loads as child */
  @Watch('getSingleProjectInfo')
  private ongetSingleProjectInfoChanged(val: any) {
    if (val && val.statusId > 1) {
      this.showProjectActions = true;
      this.projectStatus = val.statusId;
    }
  }

  private toggleWarning() {
    this.requestDialog = !this.requestDialog;
  }

  private toggleDelete() {
    this.deleteDialog = !this.deleteDialog;
  }

  private confirmLiveAccess() {
    this.updateProjectStatus({
      projectId: this.id,
      statusId: projectStatus.developmentComplete
    });
  }
  private confirmDeleteProject() {
    this.deleteProject({ projectId: this.id });
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';

.btn-delete {
  color: $BCgovBlue10;
  cursor: pointer;
  & .icon-delete {
    color: $BCgovBlue10;
  }
}
.pad-0 {
  padding: 0 12px;
}
</style>
