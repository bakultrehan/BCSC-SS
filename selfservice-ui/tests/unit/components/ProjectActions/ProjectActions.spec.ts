import { mount } from '@vue/test-utils';
import ProjectActions from '@/components/ProjectActions/ProjectActions.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import { singleProject } from '../../mocks/projectinfo';
let vuetify: any;
vuetify = new Vuetify();
describe('ProjectActions.vue', () => {
  const store = new Vuex.Store({
    modules: {
      ProjectInfoModule: {
        namespaced: true,
        state: {},
        getters: {
          getChangeStatus: jest.fn(() => {
            return { singleProject };
          }),
          getDeleteProjectReturn: jest.fn(),
          getSingleProjectInfo: jest.fn(),
        },
        actions: {
          loadSingleProjectInfo: jest.fn(),
          deleteProject: jest.fn(),
          updateProjectStatus: jest.fn(),
        },
      },
      KeyCloakModule: {
        namespaced: true,
        state: {},
        getters: {
          isAdmin: jest.fn(() => {
            return true;
          }),
        },
        actions: { setLogout: jest.fn() },
      },
    },
  });

  const mountFunction = (options: any) => {
    return mount(ProjectActions, {
      store,
      vuetify,
      mocks: { $t: jest.fn(() => {}) }, // tslint:disable-line
      ...options,
    });
  };

  it('renders props when passed', () => {
    const wrapper = mountFunction({});
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders props on click of Live access', () => {
    const projectActionPage = mountFunction({});
    const toggleDelete = jest.fn();
    projectActionPage.setData({ projectStatus: 2 });
    const button = projectActionPage.find('.btn-req');
    projectActionPage.vm.$on('action-btn:clicked', toggleDelete);
    button.trigger('click');

    expect(projectActionPage.element).toMatchSnapshot();
  });
  it('renders props when passed with gettors', () => {
    const projectActionPage = mountFunction({});
    const toggleDelete = jest.fn();
    projectActionPage.setData({ projectStatus: 1 });

    const button = projectActionPage.find('.btn-delete');
    projectActionPage.vm.$on('action-btn:clicked', toggleDelete);
    button.trigger('click');

    expect(projectActionPage.element).toMatchSnapshot();
  });

  it('renders props on click of delete', () => {
    const projectActionPage = mountFunction({});
    const toggleDelete = jest.fn();
    projectActionPage.setData({ deleteDialog: true });
    const button = projectActionPage.find('.dialog-delete');
    projectActionPage.vm.$on('action-btn:clicked', toggleDelete);
    button.trigger('click');

    expect(projectActionPage.element).toMatchSnapshot();
  });

  it('renders props on click of delete', () => {
    const projectActionPage = mountFunction({});
    const toggleDelete = jest.fn();
    projectActionPage.setData({ requestDialog: true });
    const button = projectActionPage.find('.btn-live');
    projectActionPage.vm.$on('action-btn:clicked', toggleDelete);
    button.trigger('click');

    expect(projectActionPage.element).toMatchSnapshot();
  });
});
