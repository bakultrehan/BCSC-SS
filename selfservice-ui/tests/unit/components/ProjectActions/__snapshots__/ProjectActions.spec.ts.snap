// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectActions.vue renders props on click of Live access 1`] = `
<div>
  <div
    class="d-flex justify-end col col-12"
    no-gutters=""
  >
    <div
      class="v-card v-card--flat v-sheet theme--light"
    >
      <button
        class="btn btn-req v-btn v-btn--contained theme--light v-size--default secondary-btn normal"
        name="button"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          
        </span>
      </button>
    </div>
  </div>
   
  <div
    class="d-flex justify-end btn-delete pad-0 col"
    no-gutters=""
  >
    <i
      aria-hidden="true"
      class="v-icon notranslate ml-2 icon-delete mdi mdi-delete theme--light"
      style="font-size: 16px;"
    />
    
    
  
  </div>
   
  <div
    class="col"
  >
    <div
      class="v-dialog__container"
      role="dialog"
    >
      <div
        class="v-dialog__content v-dialog__content--active"
        role="document"
        style="z-index: 202;"
        tabindex="0"
      >
        <div
          class="v-dialog v-dialog--active v-dialog--persistent"
          style="max-width: 70%;"
        >
          <div
            class="v-card v-sheet theme--light"
          >
            <header
              class="bc-subtitle v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat"
              style="height: 64px;"
            >
              <div
                class="v-toolbar__content"
                style="height: 64px;"
              >
                <div
                  class="v-toolbar__title"
                >
                  
            
          
                </div>
              </div>
            </header>
             
            <div
              class="v-card__text text-left ma-8"
            />
             
            <div
              class="v-card__actions pb-10"
            >
              <div
                class="spacer"
              />
               
              <button
                class="btn v-btn v-btn--flat v-btn--text theme--light v-size--default secondary-btn normal"
                name="button"
                text=""
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
            
          
                </span>
              </button>
               
              <button
                class="btn btn-live v-btn v-btn--flat v-btn--text theme--light v-size--default primary normal"
                name="button"
                text=""
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
            
          
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-dialog__container"
      role="dialog"
    >
      <div
        class="v-dialog__content"
        role="document"
        style="z-index: 0;"
      >
        <div
          class="v-dialog v-dialog--persistent"
          style="max-width: 70%; display: none;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ProjectActions.vue renders props on click of delete 1`] = `
<div>
  <div
    class="d-flex justify-end col col-12"
    no-gutters=""
  >
    <div
      class="v-card v-card--flat v-sheet theme--light"
    >
      <button
        aria-disabled="true"
        class="btn btn-req v-btn v-btn--contained v-btn--disabled theme--light v-size--default secondary-btn normal disabled"
        disabled="disabled"
        name="button"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          
        </span>
      </button>
    </div>
  </div>
   
  <div
    class="d-flex justify-end btn-delete pad-0 col"
    no-gutters=""
  >
    <i
      aria-hidden="true"
      class="v-icon notranslate ml-2 icon-delete mdi mdi-delete theme--light"
      style="font-size: 16px;"
    />
    
    
  
  </div>
   
  <div
    class="col"
  >
    <div
      class="v-dialog__container"
      role="dialog"
    >
      <div
        class="v-dialog__content"
        role="document"
        style="z-index: 0;"
      >
        <div
          class="v-dialog v-dialog--persistent"
          style="max-width: 70%; display: none;"
        />
      </div>
    </div>
     
    <div
      class="v-dialog__container"
      role="dialog"
    >
      <div
        class="v-dialog__content v-dialog__content--active"
        role="document"
        style="z-index: 202;"
        tabindex="0"
      >
        <div
          class="v-dialog v-dialog--active v-dialog--persistent"
          style="max-width: 70%;"
        >
          <div
            class="v-card v-sheet theme--light"
          >
            <header
              class="bc-subtitle v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat"
              style="height: 64px;"
            >
              <div
                class="v-toolbar__content"
                style="height: 64px;"
              >
                <div
                  class="v-toolbar__title"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate ml-2 mdi mdi-delete theme--dark"
                    medium=""
                  />
                  
            
          
                </div>
              </div>
            </header>
             
            <div
              class="v-card__text text-left ma-8"
            />
             
            <div
              class="v-card__actions pb-10"
            >
              <div
                class="spacer"
              />
               
              <button
                class="btn v-btn v-btn--flat v-btn--text theme--light v-size--default secondary-btn normal"
                name="button"
                text=""
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
            
          
                </span>
              </button>
               
              <button
                class="btn dialog-delete v-btn v-btn--flat v-btn--text theme--light v-size--default primary normal"
                name="button"
                text=""
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProjectActions.vue renders props on click of delete 2`] = `
<div>
  <div
    class="d-flex justify-end col col-12"
    no-gutters=""
  >
    <div
      class="v-card v-card--flat v-sheet theme--light"
    >
      <button
        aria-disabled="true"
        class="btn btn-req v-btn v-btn--contained v-btn--disabled theme--light v-size--default secondary-btn normal disabled"
        disabled="disabled"
        name="button"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          
        </span>
      </button>
    </div>
  </div>
   
  <div
    class="d-flex justify-end btn-delete pad-0 col"
    no-gutters=""
  >
    <i
      aria-hidden="true"
      class="v-icon notranslate ml-2 icon-delete mdi mdi-delete theme--light"
      style="font-size: 16px;"
    />
    
    
  
  </div>
   
  <div
    class="col"
  >
    <div
      class="v-dialog__container"
      role="dialog"
    >
      <div
        class="v-dialog__content v-dialog__content--active"
        role="document"
        style="z-index: 202;"
        tabindex="0"
      >
        <div
          class="v-dialog v-dialog--active v-dialog--persistent"
          style="max-width: 70%;"
        >
          <div
            class="v-card v-sheet theme--light"
          >
            <header
              class="bc-subtitle v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat"
              style="height: 64px;"
            >
              <div
                class="v-toolbar__content"
                style="height: 64px;"
              >
                <div
                  class="v-toolbar__title"
                >
                  
            
          
                </div>
              </div>
            </header>
             
            <div
              class="v-card__text text-left ma-8"
            />
             
            <div
              class="v-card__actions pb-10"
            >
              <div
                class="spacer"
              />
               
              <button
                class="btn v-btn v-btn--flat v-btn--text theme--light v-size--default secondary-btn normal"
                name="button"
                text=""
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
            
          
                </span>
              </button>
               
              <button
                class="btn btn-live v-btn v-btn--flat v-btn--text theme--light v-size--default primary normal"
                name="button"
                text=""
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
            
          
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-dialog__container"
      role="dialog"
    >
      <div
        class="v-dialog__content"
        role="document"
        style="z-index: 0;"
      >
        <div
          class="v-dialog v-dialog--persistent"
          style="max-width: 70%; display: none;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ProjectActions.vue renders props when passed 1`] = `
<div>
  <div
    class="d-flex justify-end col col-12"
    no-gutters=""
  >
    <div
      class="v-card v-card--flat v-sheet theme--light"
    >
      <button
        aria-disabled="true"
        class="btn btn-req v-btn v-btn--contained v-btn--disabled theme--light v-size--default secondary-btn normal disabled"
        disabled="disabled"
        name="button"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          
        </span>
      </button>
    </div>
  </div>
   
  <div
    class="d-flex justify-end btn-delete pad-0 col"
    no-gutters=""
  >
    <i
      aria-hidden="true"
      class="v-icon notranslate ml-2 icon-delete mdi mdi-delete theme--light"
      style="font-size: 16px;"
    />
    
    
  
  </div>
   
  <!---->
</div>
`;

exports[`ProjectActions.vue renders props when passed with gettors 1`] = `
<div>
  <div
    class="d-flex justify-end col col-12"
    no-gutters=""
  >
    <div
      class="v-card v-card--flat v-sheet theme--light"
    >
      <button
        aria-disabled="true"
        class="btn btn-req v-btn v-btn--contained v-btn--disabled theme--light v-size--default secondary-btn normal disabled"
        disabled="disabled"
        name="button"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          
        </span>
      </button>
    </div>
  </div>
   
  <div
    class="d-flex justify-end btn-delete pad-0 col"
    no-gutters=""
  >
    <i
      aria-hidden="true"
      class="v-icon notranslate ml-2 icon-delete mdi mdi-delete theme--light"
      style="font-size: 16px;"
    />
    
    
  
  </div>
   
  <div
    class="col"
  >
    <div
      class="v-dialog__container"
      role="dialog"
    >
      <div
        class="v-dialog__content"
        role="document"
        style="z-index: 0;"
      >
        <div
          class="v-dialog v-dialog--persistent"
          style="max-width: 70%; display: none;"
        />
      </div>
    </div>
     
    <div
      class="v-dialog__container"
      role="dialog"
    >
      <div
        class="v-dialog__content v-dialog__content--active"
        role="document"
        style="z-index: 202;"
        tabindex="0"
      >
        <div
          class="v-dialog v-dialog--active v-dialog--persistent"
          style="max-width: 70%;"
        >
          <div
            class="v-card v-sheet theme--light"
          >
            <header
              class="bc-subtitle v-sheet v-sheet--tile theme--dark v-toolbar v-toolbar--flat"
              style="height: 64px;"
            >
              <div
                class="v-toolbar__content"
                style="height: 64px;"
              >
                <div
                  class="v-toolbar__title"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate ml-2 mdi mdi-delete theme--dark"
                    medium=""
                  />
                  
            
          
                </div>
              </div>
            </header>
             
            <div
              class="v-card__text text-left ma-8"
            />
             
            <div
              class="v-card__actions pb-10"
            >
              <div
                class="spacer"
              />
               
              <button
                class="btn v-btn v-btn--flat v-btn--text theme--light v-size--default secondary-btn normal"
                name="button"
                text=""
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
            
          
                </span>
              </button>
               
              <button
                class="btn dialog-delete v-btn v-btn--flat v-btn--text theme--light v-size--default primary normal"
                name="button"
                text=""
                type="button"
              >
                <span
                  class="v-btn__content"
                >
                  
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
